### ISSUE TITLE
AWS Lambda: Best way to pass columns to read_parquet without getting the "Calling the Invoke API Action failed with the message: Network Error"

### ISSUE THREAD

**Issue by harshanalla** [awslabs/aws-data-wrangler#497]
Hello!

I have a lambda application that needs to read parquet files from various buckets and create reports/
I am reading ~ 6 different tables with over 200 parquet file partitions. 
I only need ~ 10 columns of the 180 columns available in a single parquet files (and there are 6 such files), so I have tried using "columns" parameter in [read_parquet](https://aws-data-wrangler.readthedocs.io/en/stable/stubs/awswrangler.s3.read_parquet.html#awswrangler-s3-read-parquet) to save on memory and time.

This is causing the Lambda to error out and give me this "Calling the Invoke API Action failed with the message: Network Error", I think the call function is accessing the same partition multiple times due to the "column" parameter and [memory size](https://github.com/awslabs/aws-data-wrangler/blob/41c0ebbeeb751329e40f16de7214a3e0ad3143b5/awswrangler/s3/_read_parquet.py#L356).

I am working around this by reading the parquet files (with all columns) and then dropping the unwanted columns. Is there any way I can do this through AWS data wrangler (and save memory allocation)?

*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*

**Comment by igorborgest** [comment]
You can configure the s3 access block size globally through `s3_block_size` ([Related tutorial](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/021%20-%20Global%20Configurations.ipynb)).
And `wr.config.s3_block_size = -1` will avoid multiple access to the same files.

But even though, this error is weird, the default behavior (20MB block size when `columns` is given) should help you to avoid memory consumption. Are you sure you are running out of memory? Did you check into the lambda execution log?

**Comment by harshanalla** [comment]
I did not know about the Global Configurations, I will definitely play around with it, thank you!

The Lambda execution logs show that the point of failure was at `read parquet`. I have set my dev Lambda memory to 10GB and I still see this error (with `columns `passed through read_parquet) and it is not due to running out of memory. I am being charged for <500 MB of data (there are other operations taking place in the lambda function.  
I forgot to mention that I am accessing the files by assuming a role and passing the session through ` boto3_session`. I don't know if this makes a difference.
